flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializer.java

```diff
@@ -225,7 +225,7 @@ public T copy(T from) {
 		if (actualType == clazz) {
 			T target;
 			try {
-				target = (T) from.getClass().newInstance();
+				target = (T) from.getClass().getDeclaredConstructor().newInstance();
```

Comment:
this change may have performance implications as noted in the mail thread linked in the jira.